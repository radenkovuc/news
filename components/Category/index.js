import ScrollMenu from 'react-horizontal-scrolling-menu'
import ArticleCard from '../ArticleCard'
import React from 'react'
import styled from 'styled-components'

const CategoryContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: 5px 20px;
`

const CategoryName = styled.div`
  display: flex;
  margin: 15px;
  font-size: 18px;
  font-family: 'Nunito Sans black', sans-serif;
`

const Category = ({name, articles}) => {
  const Arrow = ({text}) => {
    return <div>{text}</div>
  }

  const ArrowLeft = Arrow({text: '<'})
  const ArrowRight = Arrow({text: '>'})

  const renderItems = (articles) => {
    return (
      articles &&
      articles.map((article, index) => {
        return <ArticleCard key={index} article={article} />
      })
    )
  }
  return (
    <CategoryContainer>
      <CategoryName>{name}</CategoryName>
      <ScrollMenu
        data={renderItems(articles)}
        arrowLeft={ArrowLeft}
        arrowRight={ArrowRight}
        wheel={false}
        alignCenter={false}
      />
    </CategoryContainer>
  )
}

export default Category
