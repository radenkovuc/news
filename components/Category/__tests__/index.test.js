import React from 'react'
import renderer from 'react-test-renderer'

import {createTextComponent} from '../../../test/TestUtil'
import {mockedArticles} from '../../../test/MockedData'

import Category from '../'

describe('<Category />', () => {
  it('match snapshot default', () => {
    const tree = renderer.create(createTextComponent(<Category />)).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('match snapshot with articles', () => {
    const tree = renderer
      .create(createTextComponent(<Category articles={mockedArticles} />))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
