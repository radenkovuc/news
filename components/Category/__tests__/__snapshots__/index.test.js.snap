// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> match snapshot default 1`] = `
<div
  className="Container-fiywul-0 fpjmCq"
>
  <div
    className="TitleContainer-fiywul-1 mcDnB"
  >
    <div
      className="Title-fiywul-2 bqCknc"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      
    </div>
    <div
      className="ShowHideIcon-fiywul-3 dtOJXq"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`<Category /> match snapshot with articles 1`] = `
<div
  className="Container-fiywul-0 fpjmCq"
>
  <div
    className="TitleContainer-fiywul-1 mcDnB"
  >
    <div
      className="Title-fiywul-2 bqCknc"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      
    </div>
    <div
      className="ShowHideIcon-fiywul-3 dtOJXq"
      onClick={[Function]}
    />
  </div>
</div>
`;
